addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -50,6 +84,30 @@ function ComposerFactory({ Entity }) {
             });
         }
 
+        /**
+         * @param mentionKeyword
+         */
+        insertSuggestedPartner(mentionKeyword) {
+            const selection = this.suggestedPartners.find(suggestedPartner => suggestedPartner === this.activePartner);
+            const cursorPosition = this.textInputCursorStart;
+            const textLeft = this.textInputContent.substring(0, cursorPosition - (mentionKeyword.length));
+            const textRight = this.textInputContent.substring(cursorPosition, this.textInputContent.length);
+            const insert = selection.name.replace(/ /g, '\u00a0');
+            this.update({
+                suggestedPartners: [['unlink-all']],
+                textInputContent: textLeft + insert + ' ' + textRight,
+                textInputCursorStart: textLeft.length + selection.name.length + 2,
+                textInputCursorEnd: textLeft.length + selection.name.length + 2,
```

Comment:
Why `+ 2` and not `+ 1` (1 extra for being after whitespace)? and also content uses `insert` while here we use `selection.name`, would use both `insert` for consistency.

Reply:
Done